pyk/src/pyk/testing/_kompiler.py

```diff
@@ -371,19 +371,27 @@ def hints(self, definition_dir: Path, kompile: Kompiler) -> bytes:
         return hints_file.read_bytes()
 
     @pytest.fixture(scope='class')
-    def header(self, definition_dir: Path) -> kore_header:
+    def header(self, definition_dir: Path) -> KoreHeader:
         process = subprocess.run(['kore-rich-header', str(definition_dir / 'definition.kore')], stdout=subprocess.PIPE)
         hdr = process.stdout
-        path = str(definition_dir / 'header.bin')
+        header_file_name = definition_dir.name + '.header'
+        path = definition_dir / header_file_name
         with open(path, 'wb') as f:
             f.write(hdr)
-        from ..kllvm.hints.prooftrace import kore_header
+        from ..kllvm.hints.prooftrace import KoreHeader
 
-        return kore_header(path)
+        return KoreHeader.create(path)
 
     @pytest.fixture(scope='class')
     def definition_file(self, definition_dir: Path) -> str:
         definition_path = definition_dir / 'definition.kore'
         assert definition_path.is_file()
         with open(definition_path) as f:
             return f.read()
+
+    @pytest.fixture(scope='class')
+    def hints_file(self, definition_dir: Path, kompile: Kompiler) -> Path:
+        input_kore_file = kompile._cache_definition(self.HINTS_INPUT_KORE)
+        hints_file_name = definition_dir.name.replace('.k', '.hints')
```

Comment:
This looks a bit hacky. Consider implementing a dedicated cache of hints inputs for `ProofTraceTest`. Then no renaming is needed, and `kompiler` can be dropped as a direct dependency.

(After the renaming, the cache might no longer point to the right file.)

Reply:
The renaming is just for creating a new file with a custom name instead of simply ‘hints.bin’ or something static like this. This is mainly to avoid the use of 2 different tests appending in the same file (an issue that I found while writing those tests).